application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -4182,6 +4201,41 @@ input PlanTransitModesInput {
   transit: [PlanTransitModePreferenceInput!]
 }
 
+"""
+A via-location is used to specifying a location as an intermediate place the router must
+route through. The via-location is either a pass-through-location or a visit-via-location.
+"""
+input PlanViaLocationInput @oneOf {
+  "Board, alight or pass-through(on-board) at the stop location."
+  passThrough: PlanPassThroughViaLocationInput
+  "Board or alight at a stop location or visit a coordinate."
+  visit: PlanVisitViaLocationInput
+}
+
+"""
+A visit-via-location is a physical visit to one of the stop locations or coordinates listed. An
+on-board visit does not count, the traveler must alight or board at the given stop for it to to
+be accepted. To visit a coordinate, the traveler must walk(bike or drive) to the closest point
+in the street network from a stop and back to another stop to join the transit network.
+
+NOTE! Coordinates are NOT supported yet.
+"""
+input PlanVisitViaLocationInput {
+  "The label/name of the location. This is pass-through information and is not used in routing."
+  label: String
+  """
+  The minimum wait time is used to force the trip to stay the given duration at the
+  via-location before the trip is continued.
```

Comment:
Maybe we should avoid using the word trip here due to different meaning in GTFS.

Reply:
I've used itinerary even though I'm not sure if you can "continue and itinerary".